### PR TITLE
feat: add ccp config map settings for public preview

### PR DESCRIPTION
**Pull request by bragi92**

### CODE REVIEW

File: otelcollector/configmaps/ama-metrics-settings-configmap.yaml [hunk 1]
```diff
@@ -24,6 +24,11 @@ data:
     networkobservabilityHubble = true
     networkobservabilityCilium = true
     prometheuscollectorhealth = false
+    controlplane-apiserver = true
```
Comment (bragi92) [inline otelcollector/configmaps/ama-metrics-settings-configmap.yaml:27]: - Add a link to control plan metrics doc
- Update our documentation for minimal ingestion profile

Comment (bragi92) [reply otelcollector/configmaps/ama-metrics-settings-configmap.yaml:27]: update ccp configs to add 1024 limit

Comment (bragi92) [reply otelcollector/configmaps/ama-metrics-settings-configmap.yaml:27]: doc update for minimal ingestion profile -> https://github.com/MicrosoftDocs/azure-docs-pr/pull/264369/files

File: otelcollector/configmaps/ama-metrics-settings-configmap.yaml [hunk 2]
```diff
@@ -43,6 +48,11 @@ data:
     networkobservabilityRetina = ""
     networkobservabilityHubble = ""
     networkobservabilityCilium = ""
+    controlplane-apiserver = ""
```
[the remark this answers is not in the document]
Comment (bragi92) [reply otelcollector/configmaps/ama-metrics-settings-configmap.yaml:51]: I see, this setting is to make sure that any extra metrics that they're looking to scrape gets added to the list. I am resolving this conversation right now to get this merged into main as the documentation update is going out. Let's sync in tomorrow's meeting on this.

File: otelcollector/configmaps/ama-metrics-settings-configmap.yaml [hunk 3]
```diff
@@ -24,6 +24,11 @@ data:
     networkobservabilityHubble = true
     networkobservabilityCilium = true
     prometheuscollectorhealth = false
+    controlplane-apiserver = true
+    controlplane-cluster-autoscaler = false
+    controlplane-kube-scheduler = false
+    controlplane-kube-controller-manager = false
+    controlplane-etcd = true
```
Comment (gracewehner) [inline otelcollector/configmaps/ama-metrics-settings-configmap.yaml:31]: Thanks Kaveesh, will this onboard customers to public preview if they apply this configmap? Or is there still another setting they need to enable? Or for public preview is this going to be enabled for all customers?

Comment (bragi92) [reply otelcollector/configmaps/ama-metrics-settings-configmap.yaml:31]: Here is a link to their onboarding documentation: https://learn.microsoft.com/en-us/azure/aks/monitor-control-plane-metrics


Applying this configmap will not onboard the said customers as they'll need to have enabled the feature flag.

If a customer has enabled the required feature flag on their subscription, then onboarding to Managed Prometheus will also onboard them in the underlay clusters.

Comment (gracewehner) [reply otelcollector/configmaps/ama-metrics-settings-configmap.yaml:31]: Sounds good, thanks!

**Review by gracewehner — APPROVED** [review]

**Review by Sohamdg081992 — APPROVED** [review]

**Pull request merged**